WindowsDevicePortalWrapper/WindowsDevicePortalWrapper.Shared/Core/WindowsErrorReporting.cs

```diff
@@ -25,5 +31,170 @@ public partial class DevicePortal
         /// API for getting the list of Windows error reports.
         /// </summary>
         public static readonly string WindowsErrorReportsApi = "api/wer/reports";
+
+        /// <summary>
+        /// Gets the list of Windows Error Reporting (WER) reports.
+        /// </summary>
+        /// <returns>The list of Windows Error Reporting (WER) reports.</returns>
+        public async Task<WerDeviceReports> GetWindowsErrorReportsAsync()
+        {
+            this.CheckPlatformSupport();
+
+            return await this.GetAsync<WerDeviceReports>(WindowsErrorReportsApi);
+        }
+
+        /// <summary>
+        /// Gets the list of files in a Windows Error Reporting (WER) report.
+        /// </summary>
+        /// <param name="user">The user associated with the report.</param>
+        /// <param name="type">The type of report. This can be either 'queried' or 'archived'.</param>
+        /// <param name="name">The name of the report.</param>
+        /// <returns>The list of files.</returns>
+        public async Task<WerFiles> GetWindowsErrorReportingFileListAsync(string user, string type, string name)
+        {
+            this.CheckPlatformSupport();
+
+            Dictionary<string, string> payload = new Dictionary<string, string>();
+            payload.Add("user", user);
+            payload.Add("type", type);
+            payload.Add("name", Utilities.Hex64Encode(name));
+
+            return await this.GetAsync<WerFiles>(WindowsErrorReportingFilesApi, Utilities.BuildQueryString(payload));
+        }
+
+        /// <summary>
+        /// Gets the specified file from a Windows Error Reporting (WER) report.
+        /// </summary>
+        /// <param name="user">The user associated with the report.</param>
+        /// <param name="type">The type of report. This can be either 'queried' or 'archived'.</param>
+        /// <param name="name">The name of the report.</param>
+        /// <param name="file">The name of the file to download from the report.</param>
+        /// <returns>Byte array containing the file data</returns>
+        public async Task<byte[]> GetWindowsErrorReportingFileAsync(string user, string type, string name, string file)
+        {
```

Comment:
This signature really points out how bad our REST API is - have to get fields from 2 different requests and 3 different objects to fill this in.